tests/unit/data/datasets/test_response_dataset.py

```
@@ -16,100 +16,155 @@
-        {input_key: "Bye", output_key: "Goodbye!"},
-    ]
 
-    # Create temporary files for train and validation data
-    with tempfile.NamedTemporaryFile(
-        mode="w", suffix=".json", delete=False
-    ) as train_file:
-        json.dump(train_data, train_file)
-        train_path = train_file.name
+    # Create temporary dataset file
+    if is_save_to_disk:
+        data_path = tempfile.mktemp()
```

Ruff (0.14.13)

Use of insecure and deprecated function (`mktemp`)

(S306)